partiql-ast/src/main/java/org/partiql/ast/ConflictTarget.java

```diff
@@ -0,0 +1,94 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This represents the potential targets for the ON CONFLICT clause.
+ * @see OnConflict
+ * @see OnConflict#target
+ */
+public abstract class ConflictTarget extends AstNode {
+
+    /**
+     * This is the index variant of the conflict target.
+     * @see OnConflict
+     * @see ConflictTarget
+     */
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static final class Index extends ConflictTarget {
+        // TODO: Do we like this name?
+        // TODO: Equals and hashcode
+
+        /**
+         * TODO
+         */
+        // TODO: Should this be a list of identifiers? Or paths? Expressions?
+        @NotNull
+        public final List<Identifier> indexes;
```

Comment:
I discussed with @am357 about this. There is ambiguity whether we want to allow for nested references right now. In the RFC, it's just an attribute name. In the future, maybe an identifier chain. Maybe arbitrary paths.

Reply:
I'd defer to @am357 about this. From our latest discussion, it's only a single non-qualified identifier. In the future, it could theoretically be an arbitrary path. We just don't know.

Reply:
Ok, we should sync on this next week. There's inconsistency w/ the RFC and/or current implementation that we should resolve before the final release.